File: kaggle_environments/envs/kore_fleets/starter_bots/ts/kore/Configuration.ts
export class Configuration {
    
    public readonly agentTimeout: number;
    public readonly startingKore: number;
    public readonly size: number;
    public readonly spawnCost: number;
    public readonly convertCost: number;
    public readonly regenRate: number;
    public readonly maxRegenCellKore: number;
    public readonly randomSeed: number;

    public constructor(rawConfiguration: string) {
        const config = JSON.parse(rawConfiguration);
        this.agentTimeout = config.agentTimeout;
        this.startingKore = config.startingKore;
        this.size = config.size;
        this.spawnCost = config.spawnCost;
        this.convertCost = config.convertCost;
        this.regenRate = config.regenRate;
        this.maxRegenCellKore = config.maxRegenCellKore;
        this.randomSeed = config.randomSeed;
    }
}

File: kaggle_environments/envs/kore_fleets/starter_bots/java/kore/Configuration.java
package kore;

public class Configuration {
    
    public final int agentTimeout;
    public final int startingKore;
    public final int size;
    public final double spawnCost;
    public final int convertCost;
    public final double regenRate;
    public final int maxRegenCellKore;
    public final int randomSeed;

    public Configuration(String rawConfiguration) {
        this.agentTimeout = KoreJson.getIntFromJson(rawConfiguration, "agentTimeout");
        this.startingKore = KoreJson.getIntFromJson(rawConfiguration, "startingKore");
        this.size = KoreJson.getIntFromJson(rawConfiguration, "size");
        this.spawnCost = KoreJson.getDoubleFromJson(rawConfiguration, "spawnCost");
        this.convertCost = KoreJson.getIntFromJson(rawConfiguration, "convertCost");
        this.regenRate = KoreJson.getDoubleFromJson(rawConfiguration, "regenRate");
        this.maxRegenCellKore = KoreJson.getIntFromJson(rawConfiguration, "maxRegenCellKore");
        this.randomSeed = KoreJson.getIntFromJson(rawConfiguration, "randomSeed");
    }
}
